external/llvm-project/mlir/include/mlir/Dialect/Async/IR/AsyncOps.td

```diff
@@ -27,6 +28,66 @@ include "mlir/Interfaces/SideEffectInterfaces.td"
 class Async_Op<string mnemonic, list<Trait> traits = []> :
     Op<AsyncDialect, mnemonic, traits>;
 
+def Async_LaunchOp :
+  Async_Op<"launch", [DeclareOpInterfaceMethods<CallOpInterface>,
+                      AttrSizedOperandSegments]> {
+  let summary = "Asynchronous launch operation";
+  let description = [{
+    The `async.launch` operation encapsulates a call to a 'kernel' func that
+    may run asynchronously from the caller. A 'async.token' is returned to
+    maintain ordered dependent execution.
+
+    The actual concurrency semantics depends on the dialect lowering to the
+    executable format. Fully sequential execution ("compute0" completes before
+    "compute1" starts) is a completely legal execution.
+
+    Because concurrent execution is not guaranteed, it is illegal to create an
```

Comment:
"Illegal" seems to imply that it will fail verification.  Is this the case?  If not, then you might prefer to say "undefined behavior".

Reply:
Yep, I'm thinking there is no check..